clippy_lints/src/unused_unit.rs

```diff
@@ -121,7 +121,7 @@ fn is_unit_expr(expr: &ast::Expr) -> bool {
 fn lint_unneeded_unit_return(cx: &EarlyContext<'_>, ty: &ast::Ty, span: Span) {
     let (ret_span, appl) = if let Ok(fn_source) = cx.sess().source_map().span_to_snippet(span.with_hi(ty.span.hi())) {
         fn_source
-            .rfind("->")
+            .rfind(" -> ")
```

Comment:
We should not assume the code is properly formatted.

This tests:
```rust
#[rustfmt::skip]
fn test()->(){}

#[rustfmt::skip]
fn test2() ->(){}

#[rustfmt::skip]
fn test3()-> (){}
```
will result in invalid Rust code:
```rust
#[rustfmt::skip]
fn test()->{}

#[rustfmt::skip]
fn test2() ->{}

#[rustfmt::skip]
fn test3()-> {}
```

Reply:
>We should not assume the code is properly formatted.

Thanks for your review! Ok.

How about this approach https://github.com/rust-lang/rust-clippy/pull/6247/commits/4c58860cbfe51049d48f4f31f76dcc18b39cfe98?  For non-formated codes, I think this isn't perfect but good because this doesn't result in invalid Rust code at least.

Reply:
I think your approach works, but in general, I'm hesitating if we should deal with this kind of problems. 

IMO the usual workflow should be to apply rustfix and then rustfmt (and if you skip rustfmt, deal with the extra space), but maybe we should add this to the UI? To be discussed :)